debian/control

```diff
@@ -100,6 +100,7 @@ Build-Depends: cmake (>= 3.10.2),
 # Make-Check   xmlstarlet,
                nasm [amd64],
                zlib1g-dev,
+# Make-Check   libffi-dev,
```

Comment:
@lnsyyj could you order it alphabetically? could you note down the reason why we need this change in the commit message?

Reply:
Okay, when I use the new ubuntu20.04 system to execute install-deps.sh, the following error appears.
But now I have not figured out why I need to add `# Make-Check`, can I not add it here?

```
Building wheels for collected packages: ceph, cffi                                                                                                                                                                                   
  Building wheel for ceph (setup.py) ... done                                                                                                                                                                                        
  Created wheel for ceph: filename=ceph-1.0.0-py3-none-any.whl size=32716 sha256=50c08104e260628d56425538bbfde2c9a922dce772027570903ed8c1a98771a9                                                                                    
  Stored in directory: /tmp/pip-ephem-wheel-cache-liv3foko/wheels/66/6f/d0/d2e83c6b8bbaf0907feffe78cddb53e5dd3ee7c250c3365ce3                                                                                                        
  Building wheel for cffi (setup.py) ... error                                                                                                                                                                                       
  ERROR: Command errored out with exit status 1:                                                                  
   command: /root/github/ceph/install-deps-python3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-wheel-g9687tiy/cffi/setup.py'"'"'; __file__='"'"'/tmp/pip-wheel-g9687tiy/cffi/setup.py'"'"';f=geta
ttr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-56ho5bri
       cwd: /tmp/pip-wheel-g9687tiy/cffi/
  Complete output (56 lines):
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libffi' found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libffi' found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libffi' found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libffi' found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libffi' found
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-aarch64-3.8
  creating build/lib.linux-aarch64-3.8/cffi
  copying cffi/pkgconfig.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/pkgconfig.py -> build/lib.linux-aarch64-3.8/cffi                                                                                                                                                                      
  copying cffi/backend_ctypes.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/commontypes.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/recompiler.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/vengine_gen.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/lock.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/ffiplatform.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/setuptools_ext.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/verifier.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/vengine_cpy.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/__init__.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/api.py -> build/lib.linux-aarch64-3.8/cffi 
  copying cffi/model.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/error.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/cffi_opcode.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/cparser.py -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/_cffi_include.h -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/parse_c_type.h -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/_embedding.h -> build/lib.linux-aarch64-3.8/cffi
  copying cffi/_cffi_errors.h -> build/lib.linux-aarch64-3.8/cffi
  running build_ext
  building '_cffi_backend' extension
  creating build/temp.linux-aarch64-3.8
  creating build/temp.linux-aarch64-3.8/c
  aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-secu
rity -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/root/github/ceph/install-deps-python3/include -I/usr/include/python3.8 -c c/_cffi_backend.c -o build/tem
p.linux-aarch64-3.8/c/_cffi_backend.o
  c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
     15 | #include <ffi.h>
        |          ^~~~~~~
  compilation terminated.
  error: command 'aarch64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for cffi
  Running setup.py clean for cffi
Successfully built ceph
Failed to build cffi
ERROR: Failed to build one or more wheels
```

Reply:
@lnsyyj i think chances are that some python packages do not offer precompiled binary packages for aarch64 on PyPI. so we didn't run into this issue on amd64. but since this only happens on non-amd64 arches, could you instead add
```
# Make-Check   libffi-dev [!amd64],
```

see https://www.debian.org/doc/debian-policy/ch-relationships.html

regarding "# Make-Check", i think it depends on if you can *build* ceph packages without libffi-dev in a clean-room environment. `# Make-Check` is just a convenient way for tracking the dependencies for running "make check".  they are not necessary for those who just want to *build* Ceph.

we could use "the restriction formula syntax" in future for adding optional Build-Depends. see https://wiki.debian.org/BuildProfileSpec. but i need to do some research first.

Reply:
Thank you @tchaikov , I learned.